### PR TITLE
fix #535 Add a message saying only 1 photo is compulsory

### LINKED ISSUE #535: Add a message saying only 1 photo is compulsory
Add a message saying only 1 photo is compulsory

"You don't have to fill all the info right now. Just 1 photo required."

### PR DESCRIPTION
**Pull request by Shubham2320**
## Description

A cardView has been implemented to show the required message in this PR along with some minor Ux changes.

## Related issues and discussion
fixes #535 
 
 ## Screen-shots, if any
 
**Before:**

**After:**

### CODE REVIEW

**Comment by teolemon** [comment]
It would be great to be able to dismiss the message (once you've read it once).

**Comment by Shubham2320** [comment]
@teolemon yeah I would add that too. :)

**Comment by Shubham2320** [comment]
Hey, @teolemon I've updated the PR now an option which allows dismissing the message has also been implemented. Kindly review it again.

**Before dismissing the message:**



**After dismissing the message:**

**Comment by Karljoones** [comment]
Could we change the icons on this? In my opinion, the green icon would look better if it was slightly larger and vertically centered with the text. The "x" button should be small and in the top right corner, it can be a greyish color so it kind of blends into the background but is visible enough for use

**Comment by Shubham2320** [comment]
Hey, @Karljoones I've updated the PR please review it.

**Screenshot:**

**Comment by Karljoones** [comment]
Thanks for doing those changes, however, I still think that it would look a little better if the "x" button was a much smaller, I just think that it takes up too much space. Other than that I think it looks great :)

**Comment by Shubham2320** [comment]
@Karljoones thanx for suggestions, I've changed the size of the cross do you still suggest any more changes?

**Screenshot: **

**Comment by Karljoones** [comment]
I'm sorry to be so picky but the padding that you previously had on the view was perfect, I would ask that you put this back as it now looks too crowded. I still think that the close button looks a little odd, this might better explain what I am trying to say
This might explain my point a little better

**Comment by Shubham2320** [comment]
@Karljoones Really now it seems much better, Thanx.

**Screenshot:**

File: app/src/main/res/values/dimens.xml [hunk 1]
```diff
@@ -21,6 +21,9 @@
     <dimen name="button_height_tall">60dp</dimen>
     <dimen name="button_height_normal">40dp</dimen>
     <dimen name="button_height_short">32dp</dimen>
+    <dimen name="button_height_too_short">20dp</dimen>
+    <dimen name="button_height_too_much_short">10dp</dimen>
+    button_height_too_much_short
```
Comment (Karljoones) [inline app/src/main/res/values/dimens.xml]: This line will cause crashing of the app :)

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -4,9 +4,10 @@
     <string name="action_about">About</string>
     <string name="action_discover">Discover</string>
     <string name="contribute">How to contribute</string>
-
     <string name="action_find">Find</string>
     <string name="text_barcode">Type the barcode :</string>
+    <string name="message_in_activity_save_product_offline">Only one photo is compulsory to process upload.</string>
+
```
Comment (teolemon) [inline app/src/main/res/values/strings.xml]: Only one photo is compulsory to send a new product.

**Comment by Shubham2320** [comment]
@teolemon , @Karljoones I've updated the changes please review. :)

**Comment by teolemon** [comment]
Can we merge @Karljoones  ?

**Comment by teolemon** [comment]
thank you @Shubham2320 :)
We'll need to file another bug to ensure the application remembers that this message was dismissed.

**Comment by Shubham2320** [comment]
@teolemon thanx for merging:). Shall I create a new issue and start working on that to ensure the application remembers that this message was dismissed and would not show it again?

**Comment by Shubham2320** [comment]
>thank you @Shubham2320 :)
We'll need to file another bug to ensure the application remembers that this message was dismissed.
Thanx for merging. 

Thanx for merging.
The issue #805 has already been created to ensure this and I'd work on that :).

**Pull request merged**